src/provider/model/AppConnection.ts

```diff
@@ -89,14 +73,22 @@ export abstract class AbstractAppWindow implements AppWindow {
     private readonly _channelContextListeners: ContextMap;
     private readonly _channelEventListeners: ChannelEventMap;
 
-    private _contextListener: boolean;
+    private _hasContextListener: boolean;
 
     private readonly _onIntentListenerAdded: Signal<[IntentType]> = new Signal();
     private readonly _onContextListenerAdded: Signal<[]> = new Signal();
     private readonly _onChannelContextListenerAdded: Signal<[ContextChannel]> = new Signal();
 
-    constructor(identity: Identity, appInfo: Application, maturePromise: Promise<void>, channel: ContextChannel, appWindowNumber: number) {
+    constructor(
+        identity: Identity,
+        entityType: EntityType,
+        appInfo: Application,
+        maturePromise: Promise<void>,
+        channel: ContextChannel,
+        appWindowNumber: number
```

Comment:
We're using the `appWindowNumber` name in a bunch of places which I feels wrong now

Reply:
Going with `entityIndex`, can discuss naming if you have other suggestions.

Reply:
Would still go with number - index suggests literal position in an array

Reply:
Would still go with number. 'number' - the number of this connect. 'counter' - the variable that counts total connections